### ISSUE TITLE
five_mem_node example panics

### ISSUE THREAD

**Issue by fin-ger** [pingcap/raft-rs#205]
**Describe the bug**
Running the [`five_mem_node`](https://github.com/pingcap/raft-rs/blob/master/examples/five_mem_node/main.rs) produces the following panic:

```
$ cargo run --example five_mem_node      
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/examples/five_mem_node`
 INFO 2019-04-05T13:12:51Z: raft::raft: peer_1 became follower at term 0
 INFO 2019-04-05T13:12:51Z: raft::raft: peer_1 newRaft [peers: [(1, Progress { matched: 1, next_idx: 2, state: Probe, paused: false, pending_snapshot: 0, recent_active: false, ins: Inflights { start: 0, count: 0, buffer: [] } })], term: 0, commit: 1, applied: 1, last_index: 1, last_term: 0, pending_membership_change: None]
 INFO 2019-04-05T13:12:52Z: raft::raft: peer_1 is starting a new election at term 0
 INFO 2019-04-05T13:12:52Z: raft::raft: peer_1 became candidate at term 1
 INFO 2019-04-05T13:12:52Z: raft::raft: 1 received MsgRequestVote from 1 at term 1
 INFO 2019-04-05T13:12:52Z: raft::raft: peer_1 became leader at term 1
 INFO 2019-04-05T13:12:52Z: raft::raft:  became follower at term 0
 INFO 2019-04-05T13:12:52Z: raft::raft:  newRaft [peers: [], term: 0, commit: 0, applied: 0, last_index: 0, last_term: 0, pending_membership_change: None]
 INFO 2019-04-05T13:12:52Z: raft::raft:  [term: 0] received a MsgHeartbeat message with higher term from 1 [term: 1]
 INFO 2019-04-05T13:12:52Z: raft::raft:  became follower at term 1
 INFO 2019-04-05T13:12:52Z: raft::raft:  [commit: 0, lastindex: 0, lastterm: 0] fast-forwarded commit to snapshot [index: 1, term: 0]
thread '<unnamed>' panicked at ' to_commit 1 is out of range [last_index 0]', src/raft_log.rs:248:13
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:200
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:215
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:478
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:385
   6: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:340
   7: <raft::raft_log::RaftLog<T>>::commit_to
             at ./<::std::macros::panic macros>:8
   8: <raft::raft::Raft<T>>::restore_raft
             at ./src/raft.rs:2011
   9: <raft::raft::Raft<T>>::restore
             at ./src/raft.rs:2065
  10: <raft::raft::Raft<T>>::handle_snapshot
             at ./src/raft.rs:1975
  11: <raft::raft::Raft<T>>::step_follower
             at ./src/raft.rs:1849
  12: <raft::raft::Raft<T>>::step
             at ./src/raft.rs:1207
  13: <raft::raw_node::RawNode<T>>::step
             at ./src/raw_node.rs:340
  14: five_mem_node::Node::step
             at examples/five_mem_node/main.rs:186
  15: five_mem_node::main::{{closure}}
             at examples/five_mem_node/main.rs:69
 WARN 2019-04-05T13:12:52Z: five_mem_node: send raft message to 2 fail, let Raft retry it
 WARN 2019-04-05T13:12:53Z: five_mem_node: send raft message to 2 fail, let Raft retry it
 WARN 2019-04-05T13:12:53Z: five_mem_node: send raft message to 2 fail, let Raft retry it
 WARN 2019-04-05T13:12:53Z: five_mem_node: send raft message to 2 fail, let Raft retry it
 WARN 2019-04-05T13:12:53Z: five_mem_node: send raft message to 2 fail, let Raft retry it
```

**To Reproduce**
1. Go to the project root
2. Run `cargo run --example five_mem_node`
3. The panic occurs

**Expected behavior**
The example does not panic.

**System information**
* CPU architecture: i7-6700HQ (amd64)
* Distribution and kernel version: custom built linux kernel
* SELinux on?: no
* Any other system details we should know?: using rustup nightly

**Comment by siddontang** [comment]
oh

PTAL @Hoverbear

**Comment by Hoverbear** [comment]
Oh, I think this is related to @hicqu 's changes recently around the initial dummy entry?

**Comment by Hoverbear** [comment]
Fixed by #214
